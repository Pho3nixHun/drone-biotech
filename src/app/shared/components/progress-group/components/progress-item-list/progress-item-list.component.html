<ul class="flex flex-col gap-y-2">
    <ng-content select="app-progress-item" />
</ul>
