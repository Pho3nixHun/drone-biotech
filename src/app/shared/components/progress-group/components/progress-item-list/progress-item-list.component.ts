import { Component } from '@angular/core';

@Component({
    selector: 'app-progress-item-list',
    imports: [],
    templateUrl: './progress-item-list.component.html',
})
export class ProgressItemListComponent {}
