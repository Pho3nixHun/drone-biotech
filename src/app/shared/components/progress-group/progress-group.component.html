<div class="flex flex-row space-x-2">
    <ng-content select="app-step-indicator" />
    <div class="w-full flex flex-col gap-y-3">
        <ng-content select="strong" />
        <div>
            <ng-content select="app-progress-item-list" />
        </div>
    </div>
</div>
