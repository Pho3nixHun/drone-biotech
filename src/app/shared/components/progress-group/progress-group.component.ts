import { Component } from '@angular/core';

@Component({
    selector: 'app-progress-group',
    imports: [],
    templateUrl: './progress-group.component.html',
})
export class ProgressGroupComponent {}
