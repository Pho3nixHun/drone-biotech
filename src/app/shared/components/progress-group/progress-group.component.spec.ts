import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ProgressGroupComponent } from './progress-group.component';
import { Component } from '@angular/core';
import { ProgressItemListComponent } from './components/progress-item-list/progress-item-list.component';

@Component({
    imports: [ProgressGroupComponent, ProgressItemListComponent],
    template: `
        <app-progress-item-group>
            <app-progress-item-list></app-progress-item-list>
        </app-progress-item-group>
    `,
})
class TestHostComponent {}
describe('ProgressGroupComponent', () => {
    let compiled: HTMLElement;
    let fixture: ComponentFixture<TestHostComponent>;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should render the template correctly', () => {
        // There is no need to arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
