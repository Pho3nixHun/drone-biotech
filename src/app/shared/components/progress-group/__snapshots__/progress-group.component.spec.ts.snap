// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressGroupComponent should render the template correctly 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-progress-item-group>
    <app-progress-item-list>
      <ul
        class="flex flex-col gap-y-2"
      />
    </app-progress-item-list>
  </app-progress-item-group>
</div>
`;
